libsawtooth/src/client/rest.rs

```diff
@@ -117,6 +94,50 @@ impl SawtoothClient for RestApiSawtoothClient {
     }
 }
 
+/// used for deserializing single objects returned by the REST API.
+fn get<T>(url: &str) -> Result<Option<T>, SawtoothClientError>
+where
+    T: for<'a> serde::de::Deserialize<'a> + Sized,
+{
+    let request = Client::new().get(url);
+    let response = request
+        .send()
+        .map_err(|err| SawtoothClientError::new_with_source("request failed", err.into()))?;
+
+    if response.status().is_success() {
+        let obj: Single<T> = response.json().map_err(|err| {
+            SawtoothClientError::new_with_source("failed to deserialize response body", err.into())
+        })?;
+        Ok(Some(obj.data))
+    } else if response.status().as_u16() == 404 {
+        Ok(None)
+    } else {
+        let status = response.status();
+        let msg: ErrorResponse = response.json().map_err(|err| {
+            SawtoothClientError::new_with_source(
+                "failed to deserialize error response body",
+                err.into(),
+            )
+        })?;
+        Err(SawtoothClientError::new(&format!(
+            "{} {} {}",
+            &get_err(url.to_string()),
+            status,
+            msg
+        )))
+    }
+}
+
+fn get_err(url: String) -> String {
+    let mut v: Vec<&str> = url.split('/').collect();
```

Comment:
The string parsing here is a bit fragile. For instance, if we added a query string to the URL, the batch/transaction ID in the error message would include the query string. I'd consider @peterschwarz's suggestion of passing an error message in to `get` that will get put in the final error message.

Reply:
I was definitely overthinking this, fixed this so now `get` is passed a string which is the specific error message containing the correct type and id of requested item.